Tests/SwiftFormatTests/PrettyPrint/Selections.swift

```diff
@@ -0,0 +1,231 @@
+  }
+
+  func testSpacesFullLine() {
+    // FIXME: we should be able to set the end range to _after_ the opening brace.
+    //   Currently, this also indents the comment line. Also, it would fail idempotency (since the
+    //   tests don't update the selection, and it would no longer end after the brace)
+    let input =
+      """
+      func foo() {
+      ➡️if let SomeReallyLongVar  =   Some.More.Stuff(), let a = myfunc() ⬅️{
+      // do stuff
+      }
+      }
+      """
+
+    let expected =
+      """
+      func foo() {
+        if let SomeReallyLongVar = Some.More.Stuff(), let a = myfunc() {
+      // do stuff
+      }
+      }
+      """
+
+    // The line length ends on the last paren of .Stuff()
+    assertPrettyPrintEqual(input: input, expected: expected, linelength: 80)
+  }
+
+  func testWrapInline() {
+    let input =
+      """
+      func foo() {
+      if let SomeReallyLongVar = ➡️Some.More.Stuff(), let a = myfunc()⬅️ {
+      // do stuff
+      }
+      }
+      """
+
+    let expected =
+      """
+      func foo() {
+      if let SomeReallyLongVar = Some.More
+          .Stuff(), let a = myfunc() {
+      // do stuff
+      }
+      }
+      """
+
+    // The line length ends on the last paren of .Stuff()
+    assertPrettyPrintEqual(input: input, expected: expected, linelength: 44)
+  }
+
+  func testCommentsOnly() {
+    let input =
+      """
+      func foo() {
+      if let SomeReallyLongVar = Some.More.Stuff(), let a = myfunc() {
+      ➡️// do stuff
+      // do more stuff⬅️
+      var i = 0
+      }
+      }
+      """
+
+    let expected =
+      """
+      func foo() {
+      if let SomeReallyLongVar = Some.More.Stuff(), let a = myfunc() {
+          // do stuff
+          // do more stuff
+      var i = 0
+      }
+      }
+      """
+
+    // The line length ends on the last paren of .Stuff()
+    assertPrettyPrintEqual(input: input, expected: expected, linelength: 80)
+  }
+
+  func testVarOnly() {
+    let input =
+      """
+      func foo() {
+      if let SomeReallyLongVar = Some.More.Stuff(), let a = myfunc() {
+      // do stuff
+      // do more stuff
+      ➡️⬅️var i = 0
+      }
+      }
+      """
+
+    let expected =
+      """
+      func foo() {
+      if let SomeReallyLongVar = Some.More.Stuff(), let a = myfunc() {
+      // do stuff
+      // do more stuff
+          var i = 0
+      }
+      }
+      """
+
+    // The line length ends on the last paren of .Stuff()
+    assertPrettyPrintEqual(input: input, expected: expected, linelength: 80)
+  }
+
+  #if false
+  func testFirstCommentAndVar() {
+    let input =
+      """
+      func foo() {
+      if let SomeReallyLongVar = Some.More.Stuff(), let a = myfunc() {
+      ➡️⬅️// do stuff
+      // do more stuff
+      ➡️⬅️var i = 0
+      }
+      }
+      """
+
+    // FIXME: this shouldn't indent the second comment
+    //    Also not idempotent since the selection doesn't get updated.
```

Comment:
Just want to leave a comment here that we should fix this and enable the test before merging the PR. It’s too easy to forget about otherwise.

Reply:
Fixed. The only issue was with idempotency. The utf8 offset in a selection range isn't valid for the formatted text. If I add support for line ranges then we can check for idempotency for formatting that won't do any line wrapping (or unwrapping).